internal/datastore/common/revisions/optimized_test.go

```diff
@@ -118,10 +122,13 @@ func TestOptimizedRevisionCache(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
 			defer cancel()
 
-			for _, expectedRev := range tc.expectedRevisions {
+			for _, expectedRevSet := range tc.expectedRevisions {
 				revision, err := or.OptimizedRevision(ctx)
 				require.NoError(err)
-				require.True(expectedRev.Equal(revision), "must return the proper revision %s != %s", expectedRev, revision)
+				printableRevSet := lo.Map(expectedRevSet, func(val datastore.Revision, index int) string {
+					return val.String()
+				})
+				require.Contains(expectedRevSet, revision, "must return the proper revision, allowed set %#v, received %s", printableRevSet, revision)
```

Comment:
Could we tweak this test to show that all of the candidates actually do get selected some % of the time?